spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -0,0 +1,356 @@
+import static org.apache.iceberg.expressions.Expressions.lessThanOrEqual;
+import static org.apache.iceberg.expressions.Expressions.not;
+import static org.apache.iceberg.expressions.Expressions.notIn;
+import static org.apache.iceberg.expressions.Expressions.notNull;
+import static org.apache.iceberg.expressions.Expressions.or;
+import static org.apache.iceberg.expressions.Expressions.startsWith;
+
+public class SparkV2Filters {
+
+  private static final Pattern BACKTICKS_PATTERN = Pattern.compile("([`])(.|$)");
+
+  private SparkV2Filters() {
+  }
+
+  private static final Map<String, Expression.Operation> FILTERS = ImmutableMap
+      .<String, Expression.Operation>builder()
+      .put("ALWAYS_TRUE", Expression.Operation.TRUE)
+      .put("ALWAYS_FALSE", Expression.Operation.FALSE)
+      .put("=", Expression.Operation.EQ)
+      .put("<=>", Expression.Operation.EQ)
+      .put(">", Expression.Operation.GT)
+      .put(">=", Expression.Operation.GT_EQ)
+      .put("<", Expression.Operation.LT)
+      .put("<=", Expression.Operation.LT_EQ)
+      .put("IN", Expression.Operation.IN)
+      .put("IS_NULL", Expression.Operation.IS_NULL)
+      .put("IS_NOT_NULL", Expression.Operation.NOT_NULL)
+      .put("AND", Expression.Operation.AND)
+      .put("OR", Expression.Operation.OR)
+      .put("NOT", Expression.Operation.NOT)
+      .put("STARTS_WITH", Expression.Operation.STARTS_WITH)
+      .build();
+
+  public static Expression convert(Predicate[] predicates) {
+    Expression expression = Expressions.alwaysTrue();
+    for (Predicate predicate : predicates) {
+      Expression converted = convert(predicate);
+      Preconditions.checkArgument(converted != null, "Cannot convert predicate to Iceberg: %s", predicate);
+      expression = Expressions.and(expression, converted);
+    }
+    return expression;
+  }
+
+  @SuppressWarnings({"checkstyle:CyclomaticComplexity", "checkstyle:MethodLength"})
+  public static Expression convert(Predicate predicate) {
+    if (checkIfPredicateValid(predicate) == null) {
+      return null;
+    }
+
+    Expression.Operation op = FILTERS.get(predicate.name());
+    if (op != null) {
+      switch (op) {
+        case TRUE:
+          return Expressions.alwaysTrue();
+
+        case FALSE:
+          return Expressions.alwaysFalse();
+
+        case IS_NULL:
+          return isNull(unquote(getAttrName(predicate.children()[0])));
+
+        case NOT_NULL:
+          return notNull(unquote(getAttrName(predicate.children()[0])));
+
+        case LT:
+          if (predicate.children()[1] instanceof LiteralValue) {
+            return lessThan(unquote(getAttrName(predicate.children()[0])),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[1]).value()));
+          } else {
+            return greaterThan(unquote(getAttrName(predicate.children()[1])),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[0]).value()));
```

Comment:
I think this kind of normalization (e.g. `1 < a => a > 1`) should be done in Spark. I used to have this, but I just found out it was removed. I will add this back in Spark. But before this is available in Spark, I will have it here.